src/gui/mzroll/mzfileio.cpp

```diff
@@ -79,7 +80,13 @@ mzSample* mzFileIO::loadSample(QString filename){
             QString sampleNumberInfo = " | Sample Number=" + QString::number(sample->sampleNumber);
             sampleName = sampleName + sampleNumberInfo;
         }
+
         sample->sampleName = string( sampleName.toLatin1().data() );
+        
+        mtxSampleId.lock();
```

Comment:
mutex is of no use when uploading a gcms dataset ?
@sahil21  @shubhra-agrawal  
did you test this with gcms data set ?( i am on mac )